alf/algorithms/algorithm.py

```diff
@@ -1041,13 +1041,12 @@ def update_with_gradient(self,
 
         if isinstance(loss_info.loss, torch.Tensor):
             loss = weight * loss_info.loss
-            # In theory, there might be unfinished GPU operations from previous
-            # stage before backward is called and unfinished operations of backward
-            # after backward() returns so that torch.cuda.synchronize() should
-            # be called to make sure we only measure the computation time for backward.
-            # In practice, we found that the measured time is almost same with
-            # or without cuda.synchronize().
-            # See https://pytorch.org/docs/stable/notes/cuda.html#asynchronous-execution
+            # Due to pytorch's async execution, there might be unfinished GPU
+            # operations from previous stage before backward is called and
+            # unfinished operations of backward after backward() returns.
+            #
+            # By default, record_time() will call torch.cuda.synchronize() to
+            # make sure we only measure the computation time for backward.
```

Comment:
Perhaps we can simply remove the comment?

Reply:
Sure. Updated.